src/tls/connection.c

```diff
@@ -179,7 +179,7 @@ get_iovecs (struct iovec *iov,
   int i = 0;
 
   debug (IOVEC, "  get_iovecs (%p, %i, %p, 0x%x, 0x%x)", iov, iov_length, buffer, start, end);
-  assert (end - start >= 0);
+  assert (end >= start);
```

Comment:
Indeed the original check was a no-op. But I wonder if this new check is really correct now -- end/start can both wrap around, so there may be cases where start > end legitimately, and the new assertion would break that. @allisonkarlitskaya , can you please review this bit?

Reply:
Martin is right.  This check was specifically intended to verify that the wrapped-arithmetic result of subtracting these two values had a positive signed value.  The rewritten code doesn't do the same thing at all.

Reply:
@allisonkarlitskaya : Does that even work, though? Is C automatically coalescing both variables from unsigned to int on substraction? That would be weird and too magic for C's standards.. I would have thought that this would actually do an unsigned substraction with wraparound, in which case the current assertion is a no-op.

Reply:
Ok, I misunderstood what this was supposed to test.
Anyway, I did small demo program before I opened this PR:
```
#include <assert.h>                                                             
int main() {                                                                    
      unsigned start = 5;                                                         
      unsigned end = 6;                                                           
      assert(end - start >= 0);                                                   
      assert(start - end >= 0);                                                   
      return 0;                                                                   
}
```
And it is passes just fine.
Also I get warning for both assert lines `comparison of unsigned expression >= 0 is always true`.

Reply:
@martinpitt: good point!

A few lines up from there I use a different assertion which should work better and also assert that the end is not *too* much bigger:

```
  return self->end - self->start <= BUFFER_SIZE;
```

And there, indeed, I take advantage of the fact that subtraction would return a negative as a large unsigned value (which would fail the check).  Probably it makes sense to rewrite this check to the same logic.

Reply:
I changed it, does it make sense now?